### PR TITLE
Create position objects

### PR DESCRIPTION
**Pull request by capozzid**
This PR adds new struct in utils_lib in order for the module to use them as it is already done by posi and webserver

### CODE REVIEW

File: software/modules/gui/web/src/components/arena/Joystick.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React, { FunctionComponent, useContext } from "react";
 import * as communicationProvider from "../CommunicationProvider";
 
-const FIX_SEED = 500;
+const FIX_SEED = 127;
```
[the remark this answers is not in the document]
Comment (jmmourier) [reply software/modules/gui/web/src/components/arena/Joystick.tsx:4]: Also I would recommand using moti, I thing it duplicate the code here.

**Review by jmmourier — CHANGES_REQUESTED** [review]
Thanks for the PR. 
it was a quick review from myself, this is moslty architecture discussion regarding 2 points.

File: software/modules/gui/web/src/components/arena/Joystick.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (capozzid) [reply software/modules/gui/web/src/components/arena/Joystick.tsx:4]: for the commit, yes but it's a small detail I don't think it really matter if this change is part of this commit.
About your second comment, I can't use c++ from a typescript file, one solution would be a json config file but regarding the delay I don't think it is a big issue

File: software/modules/utils/include/PositionOrientation.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+#ifndef POSITIONORIENTATION_H
+#define POSITIONORIENTATION_H
+
```
[the remark this answers is not in the document]
Comment (capozzid) [reply software/modules/utils/include/PositionOrientation.h:4]: I don't see any useful reason to have position as a composition of positionOrientation object (x,y,orientation should be at the same level. However I will change it and make positionOrientation extending position

[the remark this answers is not in the document]
Comment (capozzid) [reply software/modules/utils/include/PositionOrientation.h:4]: done

[the remark this answers is not in the document]
Comment (jmmourier) [reply software/modules/utils/include/PositionOrientation.h:4]: I really don't think it should be inheritance, it definitly seems composition to me.

[the remark this answers is not in the document]
Comment (capozzid) [reply software/modules/utils/include/PositionOrientation.h:4]: I see no advantage of calling position using positionOrientation.position.x_m and calling orientation using simply positionOrientation.orientation_rad

**Review by jmmourier — APPROVED** [review]
We did not agree, but that not blocking.
lets make this works !

**Review by devkgs — APPROVED** [review]

**Pull request merged**